src/off-chain/historical-data/cowEndpointSurplus.py

```diff
@@ -0,0 +1,258 @@
+"""
+EBBO Historical Data Testing via block number inputs or a single settlement hash.
+Uses CoW Endpoint provided callData.
+"""
+import json
+import requests
+from fractions import Fraction
+import directory
+
+
+class EBBOHistoricalDataTesting:
+    def __init__(self):
+        self.total_orders = 0
+        self.higher_surplus_orders = 0
+        self.total_surplus_eth = 0
+        self.file_name = str
+
+        self.solver_dict = {
+            "1Inch": [0, 0],
+            "Raven": [0, 0],
+            "0x": [0, 0],
+            "PLM": [0, 0],
+            "Quasilabs": [0, 0],
+            "Otex": [0, 0],
+            "SeaSolver": [0, 0],
+            "Laertes": [0, 0],
+            "ParaSwap": [0, 0],
+            "BalancerSOR": [0, 0],
+            "BaselineSolver": [0, 0],
+            "Legacy": [0, 0],
+            "NaiveSolver": [0, 0],
+            "DMA": [0, 0],
+            "CowDexAg": [0, 0],
+            "Barter": [0, 0],
+        }
+
+    def total_surplus(
+        self, start_block=None, end_block=None, settlement_hash=None, file_name=None
+    ):
+        self.file_name = file_name
```

Comment:
Should be removed. This function should only calculate the surplus and not print the results.